scripts/stablex/get_auction_elements.js

```diff
@@ -1,14 +1,110 @@
 const BatchExchange = artifacts.require("BatchExchange")
+const BN = require("bn.js")
+const argv = require("yargs")
+  .option("valid", {
+    type: "boolean",
+    describe: "Only show valid and future orders",
+  })
+  .option("covered", {
+    type: "boolean",
+    describe: "Only show orders that have some balance in the sell token",
+  })
+  .option("tokens", {
+    describe: "Filter only orders between the given tokens",
+  })
+  .help(false)
+  .version(false).argv
 
 const { decodeAuctionElements } = require("../../test/utilities.js")
 
+const COLORS = {
+  NONE: "\x1b[0m",
+  RED: "\x1b[31m",
+  YELLOW: "\x1b[33m",
+}
+
+const formatAmount = function(amount) {
+  const string = amount.toString()
+  if (string.length > 4) {
+    return `${string.substring(0, 2)} * 10^${string.length - 2}`
+  } else {
+    return string
+  }
+}
+
+const colorForValidFrom = function(order, currentBatchId) {
+  let color = COLORS.NONE
+  if (order.validFrom >= currentBatchId) {
+    color = COLORS.RED
+    if (order.validFrom - 5 <= currentBatchId) {
+      color = COLORS.YELLOW
+    }
+  }
+  return color
+}
+
+const colorForValidUntil = function(order, currentBatchId) {
+  let color = COLORS.NONE
+  if (order.validUntil - 5 <= currentBatchId) {
+    color = COLORS.YELLOW
+    if (order.validUntil <= currentBatchId) {
+      color = COLORS.RED
+    }
+  }
+  return color
+}
+
+const colorForRemainingAmount = function(order) {
+  if (
+    order.priceDenominator > 0 &&
+    order.remainingAmount
+      .mul(new BN(100))
+      .div(order.priceDenominator)
+      .toNumber() < 1
+  ) {
+    return COLORS.YELLOW
+  } else {
+    return COLORS.NONE
+  }
+}
```

Comment:
Should this colouring take number  of decimals of the token into consideration?

Reply:
It's using relative values (<1% of original amount) so it's agnostic to the number of decimals.